apis/core.oam.dev/common/types.go

```diff
@@ -345,9 +347,10 @@ type WorkflowStatus struct {
 	Mode        WorkflowMode `json:"mode"`
 	Message     string       `json:"message,omitempty"`
 
-	Suspend    bool `json:"suspend"`
-	Terminated bool `json:"terminated"`
-	Finished   bool `json:"finished"`
+	WaitSuspend bool `json:"waitSuspend"`
```

Comment:
```suggestion
	WaitSuspend bool `json:"durationWaiting"`
```

BTW, can we re-use the `suspend` field? Why should we have a new state here? \cc @FogDong

Reply:
The reasons are:

1.  checkWorkflowStatusMessage should set proper status message
2. ExecuteSteps must not rerun steps especially  in DAG mode 
3. err = e.run(taskRunners) in ExecuteSteps can only return err
4. if do not add waitSuspend, 
5.  types.Operation's waitSuspend can not be stored in cr, and it will be lost in new loop

Reply:
We can reuse the `suspend` field if we reuse the `WorkflowStateSuspended` as well.
I think it's more clear if we add `WorkflowStateWaitDurationWorkflowState` but might be too much to add `durationWaiting`, what's your opion? @wonderflow

Reply:
I think we should add a `suspendState` to indicate the real state of suspend. In this case, the suspendState is `waiting`, there can be other state in the future and we don't need to add more fields then.

By default, the suspendState is `manuelApprove`， other state can be:

1. waitDuration(1m32s)
2. pendingOnMetrics
3. ErrorOccurred

Reply:
That's good, we can add `suspendState` in `status.workflow`, and keep the `workflowState` to  `suspending`, but I'm worried that if we want to separate suspend from different states, we need to other fileds like `waitSuspend` (at least in this case) in the `status.workflow`, which make the struct more complicated.